sdk/src/types/block/semantic/mod.rs

```diff
@@ -105,17 +108,18 @@ impl<'a> SemanticValidationContext<'a> {
 
     ///
     pub fn validate(mut self) -> Result<Option<TransactionFailureReason>, Error> {
-        // Validation of inputs.
-        let mut has_implicit_account_creation_address = false;
-
-        self.commitment_context_input = self.transaction.context_inputs().commitment().copied();
+        self.commitment_context_input = self
+            .transaction
+            .context_inputs()
+            .commitment()
+            .map(|c| c.slot_commitment_id());
 
-        self.bic_context_input = self
+        let bic_context_inputs = self
             .transaction
             .context_inputs()
             .iter()
-            .find_map(|c| c.as_block_issuance_credit_opt())
-            .copied();
+            .filter_map(|c| c.as_block_issuance_credit_opt().map(|bic| *bic.account_id()))
+            .collect::<HashSet<_>>();
```

Comment:
I think you can just use `block_issuance_credits` method on the `ContextInputs` type.

Reply:
Indeed, changed, thanks